### PR TITLE
[WIP] Add initial support for Xperia 1 IV (PDX223)

### PR DESCRIPTION
**Pull request by Haxk20**
This platform is a bit different as it only supports overlaying DTBOs.

Copyleft also included lahaina DTS so i included it as a way to speedup later bringup of 1III
(I would prefer having 1IV booting first and then porting it)

Signed-off-by: Martin Botka <martin.botka@somainline.org>

### CODE REVIEW

File: arch/arm64/Kconfig.platforms [hunk 1]
```diff
@@ -208,6 +208,24 @@ config ARCH_LAHAINA
...
+	  which is based on QCOM WAIPIO chipset.
+	  If you enable this config,
+	  please use SONY Mobile device tree.
+
+config MACH_SONY_PDX223
+	bool "Sony Mobile PDX-223"
+	depends on ARCH_SONY_NAGARA
+	help
+	  Support for the SONY Mobile PDX-223 board
+	  which is based on SONY Nagaro platform.
```
Comment (MarijnS95) [inline arch/arm64/Kconfig.platforms:225]: `NagarA`, not `NagarO`, everywhere.

**Review by MarijnS95 — CHANGES_REQUESTED** [review]

File: arch/arm64/Kconfig.platforms (hunk 1 above)
Comment (MarijnS95) [reply arch/arm64/Kconfig.platforms:225]: Why was this marked as resolved?

Comment (jerpelea) [reply arch/arm64/Kconfig.platforms:225]: He force pushed. I could not grep the typo in the commit.

Comment (MarijnS95) [reply arch/arm64/Kconfig.platforms:225]: The typo was fixed in the `config` name, but not in the documentation which is what this comment was about:

image

**Comment by Haxk20** [comment]
Dammit. Will fix the typo when im home

**Pull request merged**